src/components/SyncContainer.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import withRouter from 'react-router-dom/withRouter';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { locationChange } from '../actions/syncActionCreators';
+
+function mapDispatchToProps(dispatch) {
+	return {
+		dispatchLocationChange: bindActionCreators(locationChange, dispatch),
+	};
+}
+
+/**
+ * @module SyncContainer
+ */
+export class SyncContainer extends React.Component {
```

Comment:
2 main reasons

1. `SyncContainer` needs to be a class with a `mapStateToProps` function, which means that it has to live _inside_ `<Provider>` - I can't add a `mapStateToProps` function directly to `PlatformApp` because it is the thing that provides the `<Provider>`. There are ways to get the same functionality without using `connect`, but they're not pretty 🙂.
2. syncing Redux state with the current route is well-defined, distinct feature of the platform app, so encapsulating it here is helpful from a modular architecture perspective - this container is essentially what the `react-router-redux` package used to do in previous versions of React Router, so it's kind of nice to still make it a standalone module.